OnBaseDocsApi/Controllers/OnBaseDocsController.cs

```diff
@@ -55,7 +59,189 @@ public IHttpActionResult GetFile(int id)
+            catch (JsonReaderException ex)
+            {
+                return BadRequestResult($"JSON parse error. {ex.Message}");
+            }
+            catch (IOException ex) when (ex.Message.Contains("MIME multipart message is not complete"))
+            {
+                return BadRequestResult(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                return InternalErrorResult(ex.Message);
+            }
+
+            if ((docStream == null) || string.IsNullOrWhiteSpace(docExt))
+                return BadRequestResult("The required parameter document content is missing.");
+            if (docAttr == null)
+                return BadRequestResult("The required parameter document attributes are missing.");
+            if (string.IsNullOrWhiteSpace(docAttr.IndexKey))
+                return BadRequestResult("The required parameter 'IndexKey' is empty.");
+            if (string.IsNullOrWhiteSpace(docAttr.DocumentType))
+                return BadRequestResult("The required parameter 'DocumentType' is empty.");
+
+            return TryHandleRequest(app =>
+            {
+                // The document is moved to staging when there are no keywords included
+                // so that we can kick off a re-index to generate the autofill keywords.
+                bool toStaging = (docAttr.Keywords == null) || !docAttr.Keywords.Any();
+
+                // When going to staging, confirm that the final doc type is actually valid.
+                DocumentType finalDocType = null;
+                if (toStaging)
+                {
+                    finalDocType = app.Core.DocumentTypes.Find(docAttr.DocumentType);
+                    if (finalDocType == null)
+                        return InternalErrorResult($"The DocumentType '{docAttr.DocumentType}' could not be found.");
+                }
+
+                var createAttr = new DocumentCreateAttributes
+                {
+                    DocumentType = toStaging ? Global.Config.StagingDocType : docAttr.DocumentType,
+                    Comment = docAttr.Comment,
+                    IndexKey = docAttr.IndexKey,
+                    Keywords = docAttr.Keywords,
+                    Ext = docExt,
+                    Stream = docStream,
+                    ToStaging = toStaging,
+                };
+                var result = CreateDocument(app, createAttr, out var doc);
+                if (result != null)
+                    return result;
+
+                if (toStaging)
+                {
+                    // We must get the document ID from this thread and not the task thread since
+                    // the document could go out of scope before the task starts.
+                    var docId = doc.ID;
+                    Task.Run(() =>
+                    {
+                        MoveDocumentToWorkflow(docId, finalDocType);
+                    });
+                }
+
+                return DocumentResult(doc);
```

Comment:
This is returning a 200 status code but we should return a 201 for a POST.

Reply:
Good catch.  Done.